pylops/utils/signalprocessing.py

```diff
@@ -307,3 +427,124 @@ def dip_estimate(
+    d: ArrayLike,
+    niter: int = 5,
+    liter: int = 20,
+    order: int = 2,
+    nsmooth: Tuple[int, int] = (10, 10),
+    damp: float = 0.0,
+    smoothing: str = "triangle",
+) -> NDArray:
+    r"""Plane-Wave Destruction (PWD) local slope estimation.
+
+    Slopes :math:`\sigma(z, x)` are estimated following the
+    plane-wave-destruction formulation (Claerbout, 1999; Fomel, 2002),
+    with optional structure-aligned smoothing preconditioning.
+
+    Parameters
+    ----------
+    d : :obj:`numpy.ndarray` or :obj:`ArrayLike`
+        Input 2D array of shape ``(nz, nx)``.
+    niter : :obj:`int`, optional
+        Number of outer PWD iterations. Default is ``5``.
+    liter : :obj:`int`, optional
+        Maximum number of inner least-squares iterations. Default is ``20``.
+    order : :obj:`int`, optional
+        Accuracy order of the all-pass filters. Use ``1`` (3-tap) or ``2`` (5-tap).
+        Default is ``2``.
+    nsmooth : :obj:`tuple` of :obj:`int`, optional
+        Smoothing lengths ``(ny, nx)`` for the preconditioner. Default ``(10, 10)``.
+    damp : :obj:`float`, optional
+        Damping factor for the least-squares solve. Default ``0.0``.
+    smoothing : :obj:`str`, optional
+        Preconditioning choice: ``"triangle"`` (default) applies a triangular
+        smoother (two boxcar passes); ``"boxcar"`` applies a single-pass boxcar.
+
+    Returns
+    -------
+    sigma : :obj:`numpy.ndarray`
+        Estimated slope field of shape ``(nz, nx)`` in samples per trace (:math:`\Delta z / \Delta x`).
+
+    Notes
+    -----
+    ``pwd_slope_estimate`` relies on kernels defined in ``_pwd2d_numba.py``.
+    When Numba is available the implementation is JIT-accelerated; otherwise a
+    pure-Python fallback is used.
+
+    References
+    ----------
+    - Claerbout, J. F. (1992). *Earth Sounding Analysis: Processing Versus Inversion*.
+    - Fomel, S. (2002). "Applications of plane-wave destruction filters."
+      *Geophysics*, 67(6), 1946-1960.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from pylops.utils.signalprocessing import pwd_slope_estimate
+    >>> nz, nx = 100, 60
+    >>> data = np.random.randn(nz, nx).astype("float32")
+    >>> sigma = pwd_slope_estimate(data, niter=3, nsmooth=(5, 5))
+    >>> sigma.shape
+    (100, 60)
+    """
+    if order not in (1, 2):
+        raise ValueError("order must be 1 (B3) or 2 (B5)")
+
+    din = np.asarray(d, dtype=np.float32, order="C")
+    if din.ndim != 2:
+        raise ValueError("input data must be 2-D")
+
+    nz, nx = din.shape
+    sigma = np.zeros((nz, nx), dtype=np.float32)
+    delta_sigma = np.zeros_like(sigma)
+    u1 = np.zeros_like(sigma)
+    u2 = np.zeros_like(sigma)
+
+    smoothing_lower = smoothing.lower()
+    if smoothing_lower == "triangle":
+        Sop = _triangular_smoothing_from_boxcars(nsmooth=nsmooth, dims=(nz, nx), dtype="float32")
+    elif smoothing_lower == "boxcar":
+        from pylops.basicoperators import Smoothing2D
```

Comment:
It's probably fine to just import this at the start since it is part of the library